source/common/router/header_parser.cc

```diff
@@ -37,16 +37,21 @@ parseHttpHeaderFormatter(const envoy::config::core::v3::HeaderValue& header_valu
   if (!Http::HeaderUtility::isModifiableHeader(key)) {
     throw EnvoyException(":-prefixed or host headers may not be modified");
   }
+  const bool raw_value = !header_value.raw_value().empty();
+  if (raw_value && !header_value.value().empty()) {
+    throw EnvoyException("cannot specify both value and raw_value");
```

Comment:
Just a thought: If both raw_value and value is configured, should we just take priority on raw_value and ignore value? Is that more tolerable for mis-configure?

Reply:
From the proto spec, raw_value sounded like a one-of promotion. I think it's better to dis-allow ambiguous cases and reject, we can always relax the rejection later.